### ISSUE TITLE
FunctionTemplates + constructors go boom

### ISSUE THREAD

**Issue by zpao** [zpao/v8monkey#77]
@sdwilsh and I did some debugging on the train... looks like `this` is null in the `CallCallback` when the function is a constructor

```
Program received signal EXC_BAD_ACCESS, Could not access memory.
Reason: KERN_INVALID_ADDRESS at address: 0x0000000000000000
0x0000000100219d02 in v8::Object::operator JSObject* (this=0x0) at v8.h:783
783   operator JSObject *() const { return JSVAL_TO_OBJECT(mVal); }
(gdb) bt
#0  0x0000000100219d02 in v8::Object::operator JSObject* (this=0x0) at v8.h:783
#1  0x0000000100217d28 in v8::Object::GetInternalField (this=0x0, index=0) at /Users/pao/Code/v8monkey/js/src/v8api/object.cpp:343
#2  0x000000010021dbe8 in v8::FunctionTemplate::CallCallback (cx=0x10090def0, argc=0, vp=0x100a00550) at /Users/pao/Code/v8monkey/js/src/v8api/functiontemplate.cpp:61
#3  0x000000010012075b in js::CallJSNative (cx=0x10090def0, native=0x10021db4c <v8::FunctionTemplate::CallCallback(JSContext*, unsigned int, jsval_layout*)>, argc=0, vp=0x100a00550) at jscntxtinlines.h:692
#4  0x000000010012080f in js::CallJSNativeConstructor (cx=0x10090def0, native=0x10021db4c <v8::FunctionTemplate::CallCallback(JSContext*, unsigned int, jsval_layout*)>, argc=0, vp=0x100a00550) at jscntxtinlines.h:711
#5  0x000000010011e4be in js::InvokeConstructor (cx=0x10090def0, argsRef=@0x7fff5fbfb1b0) at jsinterp.cpp:1286
#6  0x0000000100109742 in js::Interpret () at /Users/pao/Code/v8monkey/js/src/jsinterp.cpp:4636
#7  0x000000010011da5d in js::RunScript (cx=0x10090def0, script=0x10092bbb0, fp=0x100a004f8) at jsinterp.cpp:636
#8  0x000000010011e0d7 in js::Invoke (cx=0x10090def0, argsRef=@0x7fff5fbfc510, flags=0) at jsinterp.cpp:716
#9  0x00000001000ca256 in js_fun_apply (cx=0x10090def0, argc=2, vp=0x100a004c8) at /Users/pao/Code/v8monkey/js/src/jsfun.cpp:2216
#10 0x000000010012075b in js::CallJSNative (cx=0x10090def0, native=0x1000c9f73 <js_fun_apply(JSContext*, unsigned int, js::Value*)>, argc=2, vp=0x100a004c8) at jscntxtinlines.h:692
#11 0x000000010010a1b1 in js::Interpret () at /Users/pao/Code/v8monkey/js/src/jsinterp.cpp:4745
#12 0x000000010011da5d in js::RunScript (cx=0x10090def0, script=0x10092aa00, fp=0x100a00318) at jsinterp.cpp:636
#13 0x000000010011e0d7 in js::Invoke (cx=0x10090def0, argsRef=@0x7fff5fbfdc50, flags=0) at jsinterp.cpp:716
#14 0x00000001000c9f31 in js_fun_call (cx=0x10090def0, argc=1, vp=0x100a002e0) at /Users/pao/Code/v8monkey/js/src/jsfun.cpp:2153
#15 0x000000010012075b in js::CallJSNative (cx=0x10090def0, native=0x1000c9d35 <js_fun_call(JSContext*, unsigned int, js::Value*)>, argc=2, vp=0x100a002e0) at jscntxtinlines.h:692
#16 0x000000010010a1b1 in js::Interpret () at /Users/pao/Code/v8monkey/js/src/jsinterp.cpp:4745
#17 0x000000010011da5d in js::RunScript (cx=0x10090def0, script=0x10110d400, fp=0x100a000a8) at jsinterp.cpp:636
#18 0x000000010011e0d7 in js::Invoke (cx=0x10090def0, argsRef=@0x7fff5fbff3b0, flags=0) at jsinterp.cpp:716
#19 0x000000010011e8f0 in js::ExternalInvoke (cx=0x10090def0, thisv=@0x7fff5fbff450, fval=@0x7fff5fbff428, argc=1, argv=0x100913090, rval=0x7fff5fbff4f0) at jsinterp.cpp:839
#20 0x000000010004cac5 in JS_CallFunctionValue (cx=0x10090def0, obj=0x100e030d8, fval={asBits = 18445477440624042512, debugView = {payload47 = 4309688848, tag = JSVAL_TAG_OBJECT}, s = {payload = {i32 = 14721552, u32 = 14721552, why = 14721552, word = 18445477440624042512}}, asDouble = -nan(0xb800100e0a210), asPtr = 0xfffb800100e0a210}, argc=1, argv=0x100913090, rval=0x7fff5fbff4f0) at /Users/pao/Code/v8monkey/js/src/jsapi.cpp:5143
#21 0x000000010021be6b in v8::Function::Call (this=0x100912dc0, recv={mVal = 0x100912dd0}, argc=1, argv=0x7fff5fbff550) at /Users/pao/Code/v8monkey/js/src/v8api/function.cpp:31
#22 0x00000001000095fc in node::Load (process={mVal = 0x100912830}) at ../src/node.cc:2130
#23 0x000000010000a5ed in node::Start (argc=1, argv=<value temporarily unavailable, due to optimizations>) at ../src/node.cc:2436
#24 0x0000000100006d10 in start () at NativeX64.h:354
```

**Comment by robarnold** [comment]
I could not reproduce with a simple testcase. Don't suppose you have one?

**Comment by sdwilsh** [comment]
Rob fixed this in 084d4a544940db0ddc9d299641eeb84446e09cdb earlier.
